### PR TITLE
#49:- update description generating logic;

### PR DESCRIPTION
**Pull request by pirs1337**
Refs: #49

### CODE REVIEW

File: src/Services/SwaggerService.php [hunk 1]
```diff
@@ -365,25 +366,52 @@ protected function savePostRequestParameters($actionName, $rules, array $annotat
...
 
-    protected function saveDefinitions($objectName, $rules, array $annotations)
+    protected function saveDefinitions($objectName, $rules, $attributes, array $annotations)
     {
         $data = [
             'type' => 'object',
             'properties' => []
         ];
+
+        $uselessRules = [
```
Comment (DenTray) [inline src/Services/SwaggerService.php:380]: let's make a protected field with the fields type map and use it here and inside the `getParameterType` method

File: src/Services/SwaggerService.php [hunk 2]
```diff
@@ -365,25 +366,52 @@ protected function savePostRequestParameters($actionName, $rules, array $annotat
...
+            'boolean',
+            'string',
+            'required',
+            'numeric',
+            'digits',
+            'date'
+        ];
+
         foreach ($rules as $parameter => $rule) {
+            if (empty($annotations) && !empty($attributes)) {
```
Comment (DenTray) [inline src/Services/SwaggerService.php:392]: not sure that it required

File: src/Services/SwaggerService.php [hunk 3]
```diff
@@ -365,25 +366,52 @@ protected function savePostRequestParameters($actionName, $rules, array $annotat
...
+
             $rulesArray = (is_array($rule)) ? $rule : explode('|', $rule);
+
             $parameterType = $this->getParameterType($rulesArray);
             $this->saveParameterType($data, $parameter, $parameterType);
-            $this->saveParameterDescription($data, $parameter, $rulesArray, $annotations);
 
-            if (in_array('required', $rulesArray)) {
-                $data['required'][] = $parameter;
+            foreach ($uselessRules as $uselessRule) {
```
Comment (DenTray) [inline src/Services/SwaggerService.php:403]: could we just get diff between the rules and type-related rules?

**Review by DenTray — CHANGES_REQUESTED** [review]

File: src/Services/SwaggerService.php [hunk 4]
```diff
@@ -365,25 +380,33 @@ protected function savePostRequestParameters($actionName, $rules, array $annotat
...
-            $this->saveParameterDescription($data, $parameter, $rulesArray, $annotations);
+
+            $uselessRules = $this->rulesOfTypes;
+            $uselessRules['required'] = 'required';
 
             if (in_array('required', $rulesArray)) {
                 $data['required'][] = $parameter;
             }
+
+            $intersect = array_intersect_key(array_flip($rulesArray), $uselessRules);
```
Comment (DenTray) [inline src/Services/SwaggerService.php:406]: can we use an `arrar_diff` or `array_diff_assoc` functions?

Comment (pirs1337) [reply src/Services/SwaggerService.php:406]: I think we can use `array_diff_key`

**Review by DenTray — APPROVED** [review]

**Review by DenTray — APPROVED** [review]

**Review by DenTray — APPROVED** [review]

**Review by DenTray — APPROVED** [review]

**Review by astorozhevsky — APPROVED** [review]

**Pull request merged**